### PR TITLE
Remove deprecated JUnit related support in TestNG

### LINKED ISSUE #3022: Remove deprecated JUnit related support in TestNG
`7.9.0`

### Expected behavior

TestNG need not support JUnit anymore now.

### Is the issue reproducible on runner?

- [ ] Shell
- [x] Maven
- [x] Gradle
- [ ] Ant
- [ ] Eclipse
- [x] IntelliJ
- [ ] NetBeans

JUnit3/JUnit4 support in TestNG was deprecated as of `v7.7.0`.  This was done because the JUnit team have enhanced JUnit5 to support running TestNG using JUnit itself. So there's no need for TestNG to do the reverse of this and TestNG users can switch over to using JUnit5 if they have a mixture of JUnit and TestNG tests and need both of them to be executed by the same test runner (JUnit5 would be the driver which would internally invoke TestNG)

For more details please refer to the official documentation of the testng-engine by JUnit at https://github.com/junit-team/testng-engine

### PR DESCRIPTION
**Pull request by krmahadevan**
Closes #3022

Fixes #3022  .

### Did you remember to?

- [ ] Add test case(s)
- [X] Update `CHANGES.txt`
- [ ] Auto applied styling via `./gradlew autostyleApply`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

**Note:** For more information on contribution guidelines  please make sure you refer our [Contributing](.github/CONTRIBUTING.md) section for detailed set of steps.

### CODE REVIEW

**Comment by krmahadevan** [comment]
@juherr 
I would perhaps want to merge this maybe after some more time (maybe after a couple of weeks) so that in-case we need to do a bugfix release for `7.9.0` we could do that without this changeset creeping in.

**Review by juherr — CHANGES_REQUESTED** [review]
Lgtm but remove junit reference from dtd 1.1

**Comment by krmahadevan** [comment]
> Lgtm but remove junit reference from dtd 1.1

Took care of this as well. Once you have approved after review, I will continue to keep this PR open and rebased so that it can be merged at a later point in time (Perhaps 2nd or 3rd week of Jan).

**Review by juherr — APPROVED** [review]

File: testng-core/src/main/java/org/testng/TestNG.java
Comment (juherr) [inline testng-core/src/main/java/org/testng/TestNG.java:550]: isJUnit declaration should be removed too

**Review by juherr — APPROVED** [review]

**Pull request merged**
